src/models/Image.ts

```diff
@@ -0,0 +1,49 @@
+import { getModelForClass, modelOptions, prop } from '@typegoose/typegoose'
+import generateAndDownloadImage from '../helpers/generateAndDownloadImage'
+import uploadToIpfs from '../helpers/uploadToIpfs'
+
+@modelOptions({
+  schemaOptions: { timestamps: true },
+})
+export class Image {
+  @prop({ index: true, required: true, unique: true })
+  username!: string
+  @prop()
+  cid?: string
+  @prop({ default: true })
+  generating?: boolean
+}
+
+export const ImageModel = getModelForClass(Image)
+
+export async function findOrCreateImage(username: string): Promise<Image> {
+  let image = await ImageModel.findOne({ username })
+
+  if (!image || (!image.generating && !image.cid)) {
+    if (!image) {
+      image = new ImageModel({ generating: true, username })
+    } else {
+      image.generating = true
+    }
+    await image.save()
+
+    try {
+      const readableStream = await generateAndDownloadImage(username, 3)
+      const { cid } = await uploadToIpfs(readableStream)
+      image.cid = cid
+    } catch (e) {
+      console.error(e)
```

Comment:
I think we should report about pfp errors in sentry as well

Reply:
We should create new task for that